File: dev/node/dump.js
#!/usr/bin/env node
const fse = require('fs-extra');
const ArgumentParser = require('argparse').ArgumentParser;
const Sav = require('./Sav');

const parser = new ArgumentParser({
    description: 'Dump portion of file',
});
parser.addArgument('input', { help: 'Input file name' });
parser.addArgument('output', { help: 'Output file name' });
parser.addArgument('off', { help: 'Start offset' });
parser.addArgument('len', { help: 'Length to dump' });

const dumpData = (args) => {
    const save = new Sav(args.input);
    const ofs = +args.off;         // beginning of data to pull
    const len = +args.len;            // total number of bytes of data to pull

    fse.writeFileSync(args.output, save.getCurrent().slice(ofs, ofs + len));
};

module.exports = dumpData;

// execute if called directly from command line
if (require.main === module) {
    dumpData(parser.parseArgs());
}

File: dev/python/dump.py
#!/usr/bin/python3
import argparse
from Sav import Sav

parser = argparse.ArgumentParser(description = 'Dump portion of file')
parser.add_argument('input', help = 'Input file name')
parser.add_argument('output', help = 'Output file name')
parser.add_argument('offset', help = 'Start offset')
parser.add_argument('length', help = 'Length to dump')


def main(args):
    save = Sav(args.input)
    offset = int(args.offset, 0)
    length = int(args.length, 0)
    with open(args.output, 'wb') as fw:
        fw.write(save.getCurrent()[offset:offset+length])


if __name__ == '__main__':
    main(parser.parse_args())
